packages/pds/src/api/com/atproto/repo/uploadBlob.ts

```diff
@@ -12,9 +12,29 @@ export default function (server: Server, ctx: AppContext) {
     handler: async ({ auth, input }) => {
       const requester = auth.credentials.did
 
-      const blob = await ctx.actorStore.transact(requester, (actorTxn) => {
-        return actorTxn.repo.blob.addUntetheredBlob(input.encoding, input.body)
-      })
+      const blob = await ctx.actorStore.transact(
```

Comment:
The changes in here underline for me how it might be beneficial to split-up the db work from the upload work.  As I understand it, as long as this transaction remains open other writes to the same actor store will block/timeout.  Adding a read into the transaction makes it a little higher stakes now, as I think that kicks sqlite into doing some additional locking.

Reply:
That's a good point, I'll come back to this so that it doesn't hold the txn open while uploading 👍